indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -331,7 +328,18 @@ public static boolean runJobs(List<Jobby> jobs, HadoopDruidIndexerConfig config)
         Path workingPath = config.makeIntermediatePath();
         log.info("Deleting path[%s]", workingPath);
         try {
-          workingPath.getFileSystem(injectSystemProperties(new Configuration())).delete(workingPath, true);
+          if( workingPath.getFileSystem(injectSystemProperties(new Configuration())).delete(workingPath, true) ){
+            Iterator<Path> iterator = existing.iterator();
+            while(iterator.hasNext()) {
+              Path path = iterator.next();
+              if ( path.toString().startsWith(workingPath.toString()) ){
+                if( log.isDebugEnabled() ) {
+                  log.debug("Remove path [%s] from JobHelper.existing", path);
+                }
+                iterator.remove();
+              }
+            }
+          }
```

Comment:
1) JobHelperTest only have some test case dealing with properties.  It's not easy to find a place to write a test case.

2) I have tested its behavior in production environment.